### ISSUE TITLE
Cannot export font: TypeError: addPoint() argument after ** must be a mapping, not NoneType

### ISSUE THREAD

**Issue by Grief** [trufont/trufont#595]
Traceback (most recent call last):
  ```
File "/home/grief/github/trufont/Lib/trufont/windows/fontWindow.py", line 636, in exportFile
    self._font.export(path, format, compression=compression)
  File "/home/grief/github/trufont/Lib/trufont/objects/defcon.py", line 158, in export
    otf = func(self, **kwargs)
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/__init__.py", line 139, in compileTTF
    glyphSet = preProcessor.process()
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/preProcessor.py", line 48, in process
    func(ufo, glyphSet)
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/filters/cubicToQuadratic.py", line 43, in __call__
    modified = super(CubicToQuadraticFilter, self).__call__(font, glyphSet)
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/filters/__init__.py", line 208, in __call__
    if include(glyph) and filter_(glyph):
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/filters/cubicToQuadratic.py", line 68, in filter
    contour.drawPoints(pen)
  File "/home/grief/.local/lib/python3.6/site-packages/defcon/objects/contour.py", line 668, in drawPoints
    pointPen.endPath()
  File "/home/grief/.local/lib/python3.6/site-packages/ufoLib/pointPen.py", line 135, in endPath
    self._flushContour(segments)
  File "/home/grief/.local/lib/python3.6/site-packages/cu2qu/pens.py", line 170, in _flushContour
    self._drawPoints(new_segments)
  File "/home/grief/.local/lib/python3.6/site-packages/cu2qu/pens.py", line 223, in _drawPoints
    pen.addPoint(pt, segment_type, smooth, name, **kwargs)
TypeError: addPoint() argument after ** must be a mapping, not NoneType
```

**Comment by Grief** [comment]
That's happening only when I choose TTF for export. With OTF selected, the application starts to consume 100% of CPU for a long time, not sure if that is expected.

**Comment by anthrotype** [comment]
> the application starts to consume 100% of CPU for a long time, not sure if that is expected

yeah, that's `compreffor`, the CFF subroutines optimizer, which is known to take a lot of CPU resources (it spawns multiple sub-processes in parallel),

**Comment by anthrotype** [comment]
oh... that might actually be a bug in cu2qu's Cu2QuPointPen.
Does your font contain one of those special contours that only has off-curve points and no on-curves?
They can happen in TrueType, where all the on-curves can be "implied" at exactly half-way between the off-curves.

It's complicated to explain, but i have an idea how to solve this. Thanks for the report!

**Comment by Grief** [comment]
I've sent you an e-mail with the font which causes the TypeError around addPoint(). Redistribution of that font is forbidden by its license so please delete it after taking a look :) (if you need it, of course). Cheers!

**Comment by anthrotype** [comment]
thanks, but no need. I was able to reproduce and fix the bug.
I'll tag a new cu2qu release in next few minutes.

**Comment by anthrotype** [comment]
https://pypi.org/project/cu2qu/1.6.2/

**Comment by Grief** [comment]
@anthrotype I'm sorry, that doesn't work, please reopen:
```
Traceback (most recent call last):
  File "/home/grief/github/trufont/Lib/trufont/windows/fontWindow.py", line 636, in exportFile
    self._font.export(path, format, compression=compression)
  File "/home/grief/github/trufont/Lib/trufont/objects/defcon.py", line 158, in export
    otf = func(self, **kwargs)
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/__init__.py", line 139, in compileTTF
    glyphSet = preProcessor.process()
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/preProcessor.py", line 48, in process
    func(ufo, glyphSet)
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/filters/cubicToQuadratic.py", line 43, in __call__
    modified = super(CubicToQuadraticFilter, self).__call__(font, glyphSet)
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/filters/__init__.py", line 208, in __call__
    if include(glyph) and filter_(glyph):
  File "/home/grief/.local/lib/python3.6/site-packages/ufo2ft/filters/cubicToQuadratic.py", line 68, in filter
    contour.drawPoints(pen)
  File "/home/grief/.local/lib/python3.6/site-packages/defcon/objects/contour.py", line 668, in drawPoints
    pointPen.endPath()
  File "/home/grief/.local/lib/python3.6/site-packages/ufoLib/pointPen.py", line 135, in endPath
    self._flushContour(segments)
  File "/home/grief/.local/lib/python3.6/site-packages/cu2qu/pens.py", line 170, in _flushContour
    self._drawPoints(new_segments)
  File "/home/grief/.local/lib/python3.6/site-packages/cu2qu/pens.py", line 237, in _drawPoints
    pen.endPath()
  File "/home/grief/.local/lib/python3.6/site-packages/ufoLib/pointPen.py", line 399, in endPath
    self._flushContour()
  File "/home/grief/.local/lib/python3.6/site-packages/ufoLib/pointPen.py", line 388, in _flushContour
    pen.addPoint(pt, segmentType=segmentType, smooth=smooth, name=name, **kwargs)
  File "/home/grief/.local/lib/python3.6/site-packages/defcon/pens/glyphObjectPointPen.py", line 28, in addPoint
    self._contour.addPoint(pt, segmentType, smooth, name, identifier=identifier)
  File "/home/grief/.local/lib/python3.6/site-packages/defcon/objects/contour.py", line 641, in addPoint
    (x, y) = values
TypeError: 'NoneType' object is not iterable

```

**Comment by Grief** [comment]
Actually, it maybe a separate issue, not sure. But this error appears now when I export the same font to TTF after updating cu2qu

**Comment by anthrotype** [comment]
Ah...

**Comment by anthrotype** [comment]
@Grief please try again in a few minutes, I hope this time it should work

**Comment by anthrotype** [comment]
https://pypi.org/project/cu2qu/1.6.3/

**Comment by Grief** [comment]
Great! Thanks a lot, it exports now without an error.

**Comment by anthrotype** [comment]
pfew! thanks 👍
